File: 2014/FASE 1/JAVASCRIPT/q-erradas-js/cartas.js
function verificarOrdenacao(cartas) {
    let crescente = true;
    let decrescente = true;

    // Verificar se a sequência está ordenada crescentemente
    for (let i = 1; i < cartas.length; i++) {
        if (cartas[i] <= cartas[i - 1]) {
            crescente = false;
            break;
        }
    }

    // Verificar se a sequência está ordenada decrescentemente
    for (let i = 1; i < cartas.length; i++) {
        if (cartas[i] >= cartas[i - 1]) {
            decrescente = false;
            break;
        }
    }

    // Determinar o resultado
    if (crescente) {
        return 'C'; // Sequência ordenada crescentemente
    } else if (decrescente) {
        return 'D'; // Sequência ordenada decrescentemente
    } else {
        return 'N'; // Sequência não ordenada
    }
}

// Leitura da entrada
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('', (entrada) => {
    const cartas = entrada.split(' ').map(Number);

    // Verificar a ordenação e imprimir o resultado
    const resultado = verificarOrdenacao(cartas);
    console.log(resultado);

    rl.close();
});

File: 2014/FASE 1/C++/q-corretas-c++/cartas.cpp
#include <iostream>
using namespace std;

int main() {
    int cards[5];
    
    // Leitura das cartas
    for (int i = 0; i < 5; ++i) {
        cin >> cards[i];
    }
    
    // Verificação da ordenação
    bool crescente = true, decrescente = true;
    for (int i = 1; i < 5; ++i) {
        if (cards[i] > cards[i - 1]) {
            decrescente = false;
        } else if (cards[i] < cards[i - 1]) {
            crescente = false;
        }
    }
    
    // Saída do resultado
    if (crescente) {
        cout << "C" << endl;
    } else if (decrescente) {
        cout << "D" << endl;
    } else {
        cout << "N" << endl;
    }
    
    return 0;
}
